byway/src/main/webapp/track-interests.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+document.addEventListener('DOMContentLoaded', loadButtons());
+
+let selected = [];
+
+/** 
+ * Track the selected interests in an array according to their status. 
+ * @param {Element elem} tracks the current button element chosen.
+ */
+function checkStatus(elem) {
+  if(elem.className.includes("active")) {
+    selected.push(elem.innerText);
+  } else {
+    selected = selected.filter(function(interest) {
+      return interest !== elem.innerText;
+    });
+  }
+}
+
+/** 
+ * Update the class of the buttons to change their style.
+ * @param {Element elem} tracks the current button element chosen.
+ */
+function switchStatus(elem) {
+  let switchClass = (elem.className.includes("active")? "btn": "btn active");
+  elem.className = switchClass;
+  checkStatus(elem);
+}
+
+/** Display all the buttons onscreen with independent onClick events. */
+function loadButtons() {
+  fetch('/api/places')
+  .then(response => response.json())
+  .then((places) => {
+    let buttonSection = document.getElementById("interests");
+    places.forEach((place) => {
+      let button = setButton(place);
+      buttonSection.appendChild(button);
+    });
+  });
+}
+
+/**
+ * Creates a button element that contains the place of
+ * interest. When clicked, it switches status and indicates
+ * to the user if selected or deselected.
+ * @param {String place} contains the text of place of interest.
+ * @returns ButtonElement
+ */
+function setButton(place) {
+  let button = document.createElement("button");
+  button.innerText = place;
+  button.addEventListener('click', function(e) {
```

Comment:
Prefer arrow functions `() => ` for event listeners. Also note that you don't really need the event here, since the inner function, since you can just capture `button`. Note that because you are assigning a different event listener for each `button`, you can pass data other than the button, for example the explicit place being registered. That way you don't have to worry about synchronizing simultaneous toggling of different properties or rely on what class is set on a dom element to decide what state your data should be in.

Consider what happens if you later want to change the CSS classes under different circumstances. You could get into a situation where "active" on a button stops corresponding to whether something is in the `selected` list, and then your `checkStatus` will do the wrong thing: would add a *second* copy of the original element unnecessarily, or similar.

Treat your *data* in your code as authoritative and the DOM as merely a way of displaying that data.

So for example, you can write an event handler like: 
`button.addEventListener('click', () => toggleSelectedPlace(place, button));`
And have `toggleSelectedPlace` update the *data* in the `selected` array, and then set the classes on the button based on whether the place is actually in the array.

Reply:
I implemented the use of the arrow functions. My current implementation first checks if the selection is within the array or not, then toggles the status. I initially had it in the other order (toggle, then add) but changed it. Instead of setting the classes of the button if they're actually included, I toggled it after adding/removing a place onto the array. I still have to update the class if it were added or removed, so rather than do it individually per-action, I used it for both